### PR TITLE
[#IOPID-1818] Migrate login endpoint

### PR DESCRIPTION
**Pull request by BurnedMarshal**
⚠️ Warning
-----
Depends on https://github.com/pagopa/io-infra/pull/979

#### List of Changes

Load SPID/CIE configuration
Login endpoint
Extend `docker compose` configuration to run spid testenv
SonarCloud task in pipeline
Multifile app config

#### Motivation and Context

Migrate the logic needed to execute a login API for SPID and CIE

#### How Has This Been Tested?

unit test
local execution with docker compose

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

**Review by gquadrati — COMMENTED** [review]
Some small comments

File: apps/session-manager/src/controllers/__tests__/spid-logs.spec.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  const mockQueueClient = {
+    sendMessage: mockSendMessage,
+  } as unknown as QueueClient;
+
+  const getLoginTypeMock = vi.fn().mockReturnValue(LoginTypeEnum.LEGACY);
+
+  /* test.each`
+    finalLoginType
+    ${LoginTypeEnum.LEGACY}
+    ${LoginTypeEnum.LV}
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply apps/session-manager/src/controllers/__tests__/spid-logs.spec.ts:64]: It's a partial porting from `io-backend` but we can remove it. The logic is tested into the repository now.

File: .github/workflows/code_review.yaml [hunk 2]
```diff
@@ -44,4 +44,12 @@ jobs:
...
\ No newline at end of file
+        run: yarn code-review
+
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@19888635fa64d24e533d388efe8264402dfd721b
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        with:
+          projectBaseDir: apps/session-manager/
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply .github/workflows/code_review.yaml]: Fixed in 62d1c97

File: apps/session-manager/src/types/separated-list.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                .map((e) => e.trim())
+                .filter(Boolean)
+            : !input
+              ? [] // fallback to empty array in case of empty input
+              : input, // it should not happen, but in case we let the decoder fail
+        ),
+      String,
+    );
+
+export const CommaSeparatedListOf = GetArbitrarySeparatedListOf(",");
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply apps/session-manager/src/types/separated-list.ts]: removed in e0c1bbf

File: apps/session-manager/src/utils/feature-flag.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { FeatureFlag } from "../types/fature-flag";
+
+export const getIsUserEligibleForNewFeature =
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply apps/session-manager/src/utils/feature-flag.ts]: removed in e0c1bbf

File: .github/workflows/code_review.yaml [hunk 5]
```diff
@@ -44,4 +44,12 @@ jobs:
...
\ No newline at end of file
+        run: yarn code-review
+
+      - name: SonarCloud session-manager scan
+        uses: SonarSource/sonarcloud-github-action@19888635fa64d24e533d388efe8264402dfd721b
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        with:
+          projectBaseDir: apps/session-manager/
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply .github/workflows/code_review.yaml:55]: Will maps this activity on a specific task

File: apps/session-manager/api/internal.yaml [hunk 6]
```diff
@@ -650,6 +650,13 @@ parameters:
     description: A representation of a user data processing choice
     required: true
     x-example: DOWNLOAD
+  SpidLevel:
+    type: string
+    description: A SPID level.
+    x-extensible-enum:
+      - https://www.spid.gov.it/SpidL1
+      - https://www.spid.gov.it/SpidL2
+      - https://www.spid.gov.it/SpidL3
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply apps/session-manager/api/internal.yaml]: Fixed

File: apps/session-manager/env.example [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+NODE_TLS_REJECT_UNAUTHORIZED=0
+
+BACKEND_HOST=https://localhost
+SPID_TESTENV_URL=https://spid-testenv2:8088
+SAML_CALLBACK_URL=http://localhost:8081/assertionConsumerService
+
+CIE_METADATA_URL=https://api.is.eng.pagopa.it/idp-keys/cie/latest
+IDP_METADATA_URL=https://api.is.eng.pagopa.it/idp-keys/spid/latest
+
+SPID_LOG_STORAGE_CONNECTION_STRING=DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;QueueEndpoint=http://127.0.0.1:10001/devstoreaccount1;TableEndpoint=http://127.0.0.1:10002/devstoreaccount1;
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply apps/session-manager/env.example]: Will try this on another PR

File: apps/session-manager/package.json [hunk 8]
```diff
@@ -44,27 +46,33 @@
         "passport-http-bearer": "^1.0.1",
         "passport-strategy": "^1.0.0",
         "redis": "^4.5.1",
-        "winston": "^3.13.0"
+        "winston": "^3.13.0",
+        "xmldom": "^0.6.0"
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply apps/session-manager/package.json]: Fixed

**Review by gquadrati — APPROVED** [review]
LGTM

**Pull request merged**